server/server.js

```diff
@@ -336,3 +370,108 @@ const errorHandler = function (err, req, res, next) {
   }
 };
 app.use(errorHandler);
+
+/**
+ * For development purposes, we're running a second server on port 8001 that's
+ * used to receive webhooks. This is so we can easily expose this endpoint to
+ * the external world using ngrok without exposing the rest of our application.
+ * See this tutorial for more details on using ngrok and webhooks:
+ * https://www.youtube.com/watch?v=0E0KEAVeDyc
+ */
+
+const WEBHOOK_PORT = process.env.WEBHOOK_PORT || 8001;
+
+const webhookApp = express();
+webhookApp.use(bodyParser.urlencoded({ extended: false }));
+webhookApp.use(bodyParser.json());
+
+const webhookServer = webhookApp.listen(WEBHOOK_PORT, function () {
+  console.log(
+    `Webhook receiver is up and running at http://localhost:${WEBHOOK_PORT}/`
+  );
+});
+
+webhookApp.post("/server/receive_webhook", async (req, res, next) => {
+  try {
+    console.log("Webhook received:");
+    console.dir(req.body, { colors: true, depth: null });
+
+    // TODO: Verify webhook.
+    const product = req.body.webhook_type;
+    const code = req.body.webhook_code;
+    switch (product) {
+      case "ITEM":
+        handleItemWebhook(code, req.body);
+        break;
+      case "INCOME":
+        handleIncomeWebhook(code, req.body);
+        break;
+      default:
+        console.log(`Can't handle webhook product ${product}`);
+        break;
+    }
```

Comment:
Ohh, that's clever, but I think I prefer a switch statement here, since it's probably a little easier to understand if you're approaching the code for the first time.